pkgs/development/python-modules/hikari/default.nix

```diff
@@ -0,0 +1,79 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pytestCheckHook
+, pythonOlder
+, pythonRelaxDepsHook
+, pytest-runner
+, aiohttp
+, attrs
+, multidict
+, colorlog
+, pynacl
+, git
+, pytest
+, pytest-cov
+, pytest-randomly
+, pytest-asyncio
+, mock
+}:
+let
+  git_hash = "9ade5395ebf72350a869f559f7253fa429cad552";
+in
+buildPythonPackage rec {
+  pname = "hikari";
+  version = "2.0.0.dev120";
+
+  src = fetchFromGitHub {
+    owner = "hikari-py";
+    repo = "hikari";
+    rev = version;
```

Comment:
To remove the double logic, we can also use the git commit here instead of the variable git_hash.
the update-source-version can also set a rev by the update script.
src.ref can then be used by the substitution in patchPhase.

Possibly adding a comment that we are intentionally using a commit hash here could be useful for later.

Reply:
See https://github.com/NixOS/nixpkgs/pull/232557#discussion_r1542525480